spring-cloud-skipper-platform-cloudfoundry/src/main/java/org/springframework/cloud/skipper/deployer/cloudfoundry/CloudFoundryManifestApplicationDeployer.java

```diff
@@ -79,52 +79,49 @@ public CloudFoundryManifestApplicationDeployer(CloudFoundryApplicationManifestRe
-						if (specValues.containsKey("version")) {
-							specVersion = specValues.get("version");
-						}
-					}
-				}
-			}
-		}
-		return getCFApplicationManifest(release, cfManifestYamlString, specResource, specVersion);
-	}
+		ApplicationManifest cfApplicationManifest = CloudFoundryApplicationManifestUtils.updateApplicationName(release);
+		cfApplicationManifest = ApplicationManifest.builder()
+				.from(cfApplicationManifest)
+				.build();
 
-	private ApplicationManifest getCFApplicationManifest(Release release, String cfManifestYamlString,
-			String specResource, String specVersion) {
-		ApplicationManifest cfApplicationManifest = CFApplicationManifestUtils.updateApplicationName(cfManifestYamlString, release);
 		List<? extends CloudFoundryApplicationSkipperManifest> cfApplicationManifestList = this.cfApplicationManifestReader
 				.read(release.getManifest().getData());
 		for (CloudFoundryApplicationSkipperManifest cfApplicationSkipperManifest : cfApplicationManifestList) {
 			CloudFoundryApplicationSpec spec = cfApplicationSkipperManifest.getSpec();
-			try {
-				String resource = (StringUtils.hasText(specResource)) ? specResource : spec.getResource();
-				String version = (StringUtils.hasText(specVersion)) ? specVersion : spec.getVersion();
-				Resource application = this.delegatingResourceLoader.getResource(
-						AppDeploymentRequestFactory.getResourceLocation(resource, version));
-				cfApplicationManifest = CFApplicationManifestUtils.updateApplicationPath(cfApplicationManifest, application);
-				if (spec.getManifest() != null) {
-					cfApplicationManifest = CFApplicationManifestUtils
-							.updateApplicationManifest(cfApplicationManifest, spec.getManifest());
-				}
-			}
-			catch (Exception e) {
-				throw new IllegalArgumentException(e.getMessage());
-			}
+			String resource = spec.getResource();
+			String version = spec.getVersion();
+
+			Manifest manifest = spec.getManifest();
+			cfApplicationManifest = ApplicationManifest.builder()
+					.from(cfApplicationManifest)
+					.buildpack(manifest.getBuildpack())
+					.command(manifest.getCommand())
+					.disk(CloudFoundryApplicationManifestUtils.memoryInteger(manifest.getDiskQuota()))
+					.domains(manifest.getDomains() != null && !manifest.getDomains().isEmpty() ? manifest.getDomains()
+							: null)
+					.environmentVariables(new HashMap<>())
+					.healthCheckHttpEndpoint(manifest.getHealthCheckHttpEndpoint())
+					.healthCheckType(manifest.getHealthCheckType() != null
+							? ApplicationHealthCheck.from(manifest.getHealthCheckType().name())
+							: null)
+					.hosts(manifest.getHosts() != null && !manifest.getHosts().isEmpty() ? manifest.getHosts() : null)
+					.instances(manifest.getInstances())
+					.memory(CloudFoundryApplicationManifestUtils.memoryInteger(manifest.getMemory()))
+					.noHostname(manifest.getNoHostname())
+					.noRoute(manifest.getNoRoute())
+					.randomRoute(manifest.getRandomRoute())
+					.services(manifest.getServices())
```

Comment:
We need to have some handlers for these properties that convert properties into the appropriate type.

For instance,

When we do 
```
skipper:>release upgrade --release-name a1 --package-name logcf --package-version 1.0.1 --properties "spec.manifest.memory=2048,spec.manifest.services=mysql,rabbit"
```

The services values are expected to be updated. But, we get this:

```
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.util.ArrayList out of VALUE_STRING token
 at [Source: java.io.StringReader@7026f39b; line: 10, column: 17] (through reference chain: org.springframework.cloud.skipper.domain.CloudFoundryApplicationSkipperManifest["spec"]->org.springframework.cloud.skipper.domain.CloudFoundryApplicationSpec["manifest"]->org.springframework.cloud.skipper.domain.CloudFoundryApplicationSpec$Manifest["services"])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:270) ~[jackson-databind-2.8.11.2.jar!/:2.8.11.2]
	at com.fasterxml.jackson.databind.DeserializationContext.reportMappingException(DeserializationContext.java:1247) ~[jackson-databind-2.8.11.2.jar!/:2.8.11.2]
```

Reply:
Don't know if `spec.manifest.services=[mysql,rabbit]` would actually work. You need to remember that we're replacing yml and `mysql,rabbit` cannot be mapped to list. Essentially you're creating broken yml.

Reply:
Yeah, this did work:
```
package install --package-name logcf --release-name mylogcf --platform-name cf-dev --package-version 1.0.0 --properties "spec.manifest.memory=2048,spec.manifest.services=[mysql,rabbit]"
```

Reply:
yeah, The `upgrade` is still broken even when we specify with `[]`

```
java.lang.NullPointerException: null
	at org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryApplicationManifestUtils.getCFManifestMap(CloudFoundryApplicationManifestUtils.java:79) ~[spring-cloud-skipper-platform-cloudfoundry-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryReleaseAnalyzer.analyze(CloudFoundryReleaseAnalyzer.java:67) ~[spring-cloud-skipper-platform-cloudfoundry-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.deployer.cloudfoundry.CloudFoundryReleaseManager.createReport(CloudFoundryReleaseManager.java:146) ~[spring-cloud-skipper-platform-cloudfoundry-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.server.service.ReleaseReportService.createReport(ReleaseReportService.java:118) ~[spring-cloud-skipper-server-core-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cloud.skipper.server.service.ReleaseReportService$$FastClassBySpringCGLIB$$6bf7cb0e.invoke(<generated>) ~[spring-cloud-skipper-server-core-1.1.0.BUILD-SNAPSHOT.jar!/:1.1.0.BUILD-SNAPSHOT]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204) ~[spring-core-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:736) ~[spring-aop-4.3.18.RELEASE.jar!/:4.3.18.RELEASE]
```